projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -294,36 +294,50 @@ private function prepare_menu_item_icon( $icon ) {
 	 * @return string
 	 */
 	private function prepare_menu_item_url( $url, $parent_slug = '' ) {
-		// Calypso URLs need the base removed so they're not interpreted as external links.
-		if ( 0 === strpos( $url, 'https://wordpress.com' ) ) {
-			$url = str_replace( 'https://wordpress.com', '', $url );
-		} else {
-			$menu_hook   = get_plugin_page_hook( $url, $parent_slug );
-			$menu_file   = wp_parse_url( $url, PHP_URL_PATH ); // Removes query args to get a file name.
-			$parent_file = wp_parse_url( $parent_slug, PHP_URL_PATH );
+		// External URLS.
+		if ( preg_match( '/^https?:\/\//', $url ) ) {
+			// Allow URLs pointing to WordPress.com.
+			if ( 0 === strpos( $url, 'https://wordpress.com' ) ) {
+				// Calypso needs the domain removed so they're not interpreted as external links.
+				$url = str_replace( 'https://wordpress.com', '', $url );
+				return esc_url_raw( $url );
+			}
+
+			// Allow URLs pointing to Jetpack.com.
+			if ( 0 === strpos( $url, 'https://jetpack.com' ) ) {
```

Comment:
Do we already link to `jetpack.com` today? If so, I wonder if we should allow access to `cloud.jetpack.com` as well, to catch any links to Calypso green (even though Atomic site owners are not supposed to get such links today)?

Edit: I see now that `jetpack.com` is used when using `Automattic\Jetpack\Redirect::get_url()`; maybe it's not useful to consider `cloud.jetpack.com` then.

Reply:
> Do we already link to jetpack.com today? 

We don't, but I had to add this exception because there is test checking specifically a `jetpack.com` URL: https://github.com/Automattic/jetpack/blob/eaaa761b2a99290e31010ae778da0e34e99d1a0f/projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php#L384-L390

I wonder if that was added because for potential future instances, but now I wonder if we should just remove that, since we currently do not use any jetpack.com URL. We can always add an exception back when we really need it.

Reply:
https://github.com/Automattic/jetpack/pull/18759/commits/778e861f700232fb3926c5dfdde16d08dd0bcd0a